File: xNet/src/main/java/org/sqlunet/dom/DomFactory.java
/*
 * Copyright (c) 2022. Bernard Bou
 */

package org.sqlunet.dom;

import org.w3c.dom.Document;

import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;
import javax.xml.parsers.ParserConfigurationException;

import androidx.annotation.NonNull;

/**
 * Document factory
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public class DomFactory
{
	// D O C U M E N T

	/**
	 * Build blank document
	 *
	 * @return empty org.w3c.dom.Document
	 */
	static public Document makeDocument()
	{
		try
		{
			final DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
			factory.setNamespaceAware(true);
			final DocumentBuilder builder = factory.newDocumentBuilder();
			return builder.newDocument();
		}
		catch (@NonNull final ParserConfigurationException e)
		{
			throw new RuntimeException(e);
		}
	}
}

File: xNet/src/main/java/org/sqlunet/dom/DomFactory.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.dom

import org.w3c.dom.Document
import javax.xml.parsers.DocumentBuilderFactory
import javax.xml.parsers.ParserConfigurationException

/**
 * Document factory
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
object DomFactory {

    /**
     * Build blank document
     *
     * @return empty org.w3c.dom.Document
     */
    @JvmStatic
    fun makeDocument(): Document {
        return try {
            val factory = DocumentBuilderFactory.newInstance()
            factory.isNamespaceAware = true
            val builder = factory.newDocumentBuilder()
            builder.newDocument()
        } catch (e: ParserConfigurationException) {
            throw RuntimeException(e)
        }
    }
}
